README.md

```diff
@@ -1,123 +1,131 @@
 
-### Test cases
+See [CONTRIBUTING.md](CONTRIBUTING.md) for info on contributing to Cucumber (issues,
+PRs, etc.).
 
-Your gherkin might contain scenarios, as well as examples from tables beneath a scenario outline.
+Everyone interacting in this codebase and issue tracker is expected to follow the
+Cucumber [code of conduct](https://cucumber.io/conduct).
 
-Test cases represent the general case of both of these. We compile the AST down to instances of [`Cucumber::Core::Test::Case`](Cucumber/Core/Test/Case.html), each containing a number of instances of [`Cucumber::Core::Test::Step`](Cucumber/Core/Test/Step.html). It's these that are then filtered and executed.
+## Installation
 
-Test cases and their test steps are also immutable value objects.
+`cucumber-core` is a Ruby gem. Install it as you would install any gem: add
+`cucumber-core` to your Gemfile:
 
-### Filters
+    gem 'cucumber-core'
 
-Once we have the test cases, and they've been activated by the mappings, you may want to pass them through a filter or two. Filters can be used to do things like activate, sort, replace or remove some of the test cases or their steps before they're executed.
+then install it:
 
-### Events
+    $ bundle
 
-Events are how you find out what is happening during your test run. As the test cases and steps are executed, the runner emits events to signal what's going on.
+or install the gem directly:
 
-The following events are emitted during a run:
+    $ gem install cucumber-core
 
-- [`TestCaseStarting`](Cucumber/Core/Events/TestCaseStarting.html)
-- [`TestStepStarting`](Cucumber/Core/Events/TestStepStarting.html)
-- [`TestStepFinished`](Cucumber/Core/Events/TestStepFinished.html)
-- [`TestCaseFinished`](Cucumber/Core/Events/TestCaseFinished.html)
+### Supported platforms
 
-That's probably best illustrated with an example.
+- Ruby 3.0
+- Ruby 2.7
+- Ruby 2.6
+- Ruby 2.5
+- Ruby 2.4
+- Ruby 2.3
+- JRuby 9.2 (with [some limitations](https://github.com/cucumber/cucumber-ruby/blob/main/docs/jruby-limitations.md))
 
-## Example
+## Usage
 
-Here's an example of how you might use [`Cucumber::Core#execute`](Cucumber/Core#execute-instance_method)
+The following example aims to illustrate how to use `cucumber-core` gem and to
+make sure it is working well within your environment. For more details
+explanation on what it actually does and how to work with it, see
+[docs/ARCHITECTURE.md](docs/ARCHITECTURE.md).
 
 ```ruby
+# cucumber_core_example.rb
+
 require 'cucumber/core'
 require 'cucumber/core/filter'
 
-# This is the most complex part of the example. The filter takes test cases as input,
-# activates each step with an action block, then passes a new test case with those activated
-# steps in it on to the next filter in the chain.
 class ActivateSteps < Cucumber::Core::Filter.new
   def test_case(test_case)
     test_steps = test_case.test_steps.map do |step|
-      activate(step)
+      step.with_action { puts "processing '#{step.text}'" }
```

Comment:
We should probably be more specific and say `Kernel.puts` here just so we don't conflict with any deprecations from cucumber 5/6